model-engine/model_engine_server/infra/gateways/live_docker_image_batch_job_gateway.py

```diff
@@ -94,10 +97,27 @@ def _parse_job_status_from_k8s_obj(job: V1Job) -> BatchJobStatus:
     if status.ready is not None and status.ready > 0:
         return BatchJobStatus.RUNNING  # empirically this doesn't happen
     if status.active is not None and status.active > 0:
-        return BatchJobStatus.RUNNING  # TODO this might be a mix of pending and running
+        for pod in pods:
```

Comment:
Might be worth leaving a comment here on why a single Job resource could have multiple pods due to clarify the logic here.